flanker/addresslib/address.py

```diff
@@ -339,55 +346,34 @@ def validate_list(addr_list, as_tuple=False, metrics=False):
         >>> address.validate_address_list('a@b, c@d, e@example.com', as_tuple=True)
         ([a@mailgun.com, c@mailgun.com], ['e@example.com'])
     """
-    mtimes = {'parsing': 0, 'mx_lookup': 0,
-        'dns_lookup': 0, 'mx_conn':0 , 'custom_grammar':0}
+    mtimes = {'parsing': 0,
+              'tld_lookup': 0,
+              'mx_lookup': 0,
+              'dns_lookup': 0,
+              'mx_conn':0 ,
+              'custom_grammar':0}
 
+    # sanity check
     if not addr_list:
         return AddressList(), mtimes
 
-    # parse addresses
+    # run parser against address list
     bstart = time()
     parsed_addresses, unparseable = parse_list(addr_list, strict=True, as_tuple=True)
     mtimes['parsing'] = time() - bstart
 
     plist = AddressList()
-    ulist = []
+    ulist = unparseable
 
-    # make sure parsed list pass dns and esp grammar
```

Comment:
Why don't we need it anymore.

Reply:
Need what? The DNS and grammar checks are still there but they are no longer implemented separately for lists and individual addresses.